cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -4,11 +4,58 @@
 package integration
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
 	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
 
 	"github.com/jaegertracing/jaeger/plugin/storage/integration"
 )
 
+func createConfigWithEncoding(t *testing.T, configFile string, targetEncoding string, uniqueTopic string) string {
+	data, err := os.ReadFile(configFile)
+	require.NoError(t, err, "Failed to read config file: %s", configFile)
+
+	var config map[string]any
+	err = yaml.Unmarshal(data, &config)
+	require.NoError(t, err, "Failed to unmarshal YAML data from config file: %s", configFile)
+
+	// Function to recursively search and replace the encoding
+	var replaceEncodingAndTopic func(m map[string]any)
+	replaceEncodingAndTopic = func(m map[string]any) {
+		for k, v := range m {
+			if k == "encoding" {
+				oldEncoding := v.(string)
+				m[k] = targetEncoding
+				t.Logf("Replaced encoding '%s' with '%s' in key: %s", oldEncoding, targetEncoding, k)
```

Comment:
Instead of just logging this message I would make this function return the number of replacements that it made and assert that such number equals 2 once the top frame of that function exists. Otherwise if I change the config to not contain encoding and topic fields (eg using defaults) we'd never know that this test is not altering the config.

Reply:
Okay understood.